src/main/java/core/basesyntax/factory/Factory.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.factory;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.dao.imp.BetDaoImpl;
+import core.basesyntax.dao.imp.UserDaoImpl;
+import core.basesyntax.exceptions.MissingAnnotationException;
+import core.basesyntax.lib.Dao;
+
+public class Factory {
+    private static BetDao betDao;
+    private static UserDao userDao;
+
+    public static BetDao getBetDaoImpl() {
+        if (betDao == null) {
+            if (BetDaoImpl.class.getAnnotation(Dao.class) == null) {
```

Comment:
this check is not related to the Factory pattern. Move it to Injector

Reply:
Agree